File: framework/languages/java/e2e/SpotService/Client/src/main/java/systems/zlink/e2e/spotservice/client/Scenarios/SmA1Scenario.java
package systems.zlink.e2e.spotservice.client.Scenarios;

import systems.zlink.e2e.spotservice.shared.Contracts;

public final class SmA1Scenario extends SpotServiceScenarioContext {
    private SmA1Scenario(SpotServiceScenarioContext context) {
        super(context);
    }

    public static void run(SpotServiceScenarioContext context) {
        new SmA1Scenario(context).execute();
    }

    private void execute() {
        Contracts.StateRes first = eventually(() -> requestState("room-a", "a1", REQUEST_TIMEOUT));
        ensure("room-a".equals(first.spotRid()), "SM-A1 wrong spot rid");
        ensure("play-a".equals(first.nodeRid()), "SM-A1 wrong owner node");
        System.out.println("scenario SM-A1 passed");

    }
}

File: framework/languages/java/e2e-kotlin/SpotService/Client/src/main/kotlin/systems/zlink/e2e/kotlin/spotservice/client/scenarios/SmA1Scenario.kt
package systems.zlink.e2e.kotlin.spotservice.client.scenarios

import systems.zlink.e2e.kotlin.spotservice.client.support.ensure
import systems.zlink.e2e.kotlin.spotservice.client.support.eventually
import systems.zlink.e2e.kotlin.spotservice.client.support.SpotHttpDriver

internal object SmA1Scenario {
    suspend fun run(spots: SpotHttpDriver) {
        val first = eventually {
            spots.requestState("room-a", "a1")
        }
        ensure(first.spotRid == "room-a", "SM-A1 wrong spot rid")
        ensure(first.nodeRid == "play-a", "SM-A1 wrong owner node")
        println("scenario SM-A1 passed")
    }
}
